src/options/modules/CustomOptionTriggers.js

```diff
@@ -72,6 +72,37 @@ function applyPopupIconColor(optionValue) {
     IconHandler.changeIconIfColored(optionValue);
 }
 
+/**
+ * Adjusts UI based on the state of context menu enabled option.
+ *
+ * @function
+ * @private
+ * @param {boolean} optionValue
+ * @returns {void}
+ */
+function applyContextMenuEnabled(optionValue) {
+    if (optionValue) {
+        console.log("Context menu is enabled");
+        // Call the logic to enable the context menu
+        browser.runtime.sendMessage({ action: "enableContextMenu" });
+    } else {
+        console.log("Context menu is disabled");
+        // Call the logic to disable the context menu
+        browser.runtime.sendMessage({ action: "disableContextMenu" });
+    }
+}
+
+/**
+ * Event listener to save the checkbox state and apply it.
+ */
+document.getElementById('contextMenuEnabled').addEventListener('change', async (event) => {
+    const isChecked = event.target.checked;
+    await browser.storage.local.set({ contextMenuEnabled: isChecked });
```

Comment:
Please don't handle option saving manually.

Also, it's already saved by the library, so no need to save it again…  `AutomaticSettings.Trigger` below handles it all, so this whole custom handling/element registration should be removed…